js/api/locationObserver/index.js

```diff
@@ -0,0 +1,50 @@
+const ReactNative = require('react-native');
+const { 
+  AppState,
+} = ReactNative;
+
+// Configuration for location options, using gps, timeout and age
+const LOCATION_CONFIG = {
+  enableHighAccuracy: true, 
+  timeout: 10000, 
+  maximumAge: 1000,
+  distanceFilter: 5,
+};
+
+export default class LocationObserver {
+  constructor(onUpdate, onError) {
+    this.watchId = null;
+    this.onUpdate = onUpdate;
+    this.onError = onError;
+
+    AppState.addEventListener('change', this.onAppStateChange.bind(this));
```

Comment:
This feels a little out of scope. I don't think this class should be aware of `AppState`.

I think it might make more sense to make this class dumber and have the `<App>` take care of triggering stuff on show/hide. This way the app can attach the `AppState` listener when it's mounted (remove it when unmounted) and start/stop location updates (via a redux action) when the `AppState` changes.

This feels more testable IMO.